drivers/sensor/ina219/ina219.c

```diff
@@ -0,0 +1,381 @@
+
+		rc = ina219_reg_read(dev, INA219_REG_POWER, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading power register.");
+			return rc;
+		}
+		data->power = tmp;
+	}
+
+	if (chan == SENSOR_CHAN_ALL ||
+		chan_p == SENSOR_CHAN_INA219_BUS_CURRENT) {
+
+		rc = ina219_reg_read(dev, INA219_REG_CURRENT, &tmp);
+		if (rc) {
+			LOG_ERR("Error reading current register.");
+			return rc;
+		}
+		data->current = tmp;
+	}
+
+	return rc;
+}
+
+static int ina219_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct ina219_config *cfg = dev->config;
+	struct ina219_data *data = dev->data;
+	double tmp;
+	int8_t sign = 1;
+
+	switch ((enum sensor_channel_ina219) chan) {
+	case SENSOR_CHAN_INA219_V_SHUNT:
+		if (INA219_SIGN_BIT(data->v_shunt)) {
+			data->v_shunt = ~data->v_shunt + 1;
+			sign = -1;
+		}
+		tmp = sign * data->v_shunt * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_INA219_BUS_VOLTAGE:
+		tmp = data->v_bus * INA219_V_BUS_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_INA219_BUS_POWER:
+		tmp = data->power * cfg->current_lsb * INA219_POWER_MUL * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	case SENSOR_CHAN_INA219_BUS_CURRENT:
+		if (INA219_SIGN_BIT(data->current)) {
+			data->current = ~data->current + 1;
+			sign = -1;
+		}
+		tmp = sign * data->current * cfg->current_lsb * INA219_SI_MUL;
+		sensor_value_from_double(val, tmp);
+		break;
+	default:
+		LOG_DBG("Channel not supported by device!");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int ina219_set_power_state(const struct device *dev,
+				  uint32_t power_state)
+{
+	struct ina219_data *data = dev->data;
+	uint8_t reg_val;
+
+	if (data->pm_state == power_state) {
+		LOG_DBG("Device already in requested PM_STATE.");
+		return 0;
+	}
+
+	if (power_state == PM_DEVICE_STATE_ACTIVE) {
+		return ina219_init(dev);
+	} else if (power_state == PM_DEVICE_STATE_SUSPEND) {
+		reg_val = INA219_MODE_SLEEP;
+	} else if (power_state == PM_DEVICE_STATE_OFF) {
+		reg_val = INA219_MODE_OFF;
+	} else {
+		return 0;
+	}
+
+	if (ina219_reg_field_update(dev,
+				INA219_REG_CONF,
+				INA219_MODE_MASK,
+				reg_val)) {
+		LOG_DBG("Failed to set power state.");
+		return -EIO;
+	}
+
+	data->pm_state = power_state;
+
+	return 0;
+}
+
+static uint32_t ina219_get_power_state(const struct device *dev,
+		enum pm_device_state *state)
+{
+	struct ina219_data *data = dev->data;
+
+	*state = data->pm_state;
+
+	return 0;
+}
+
+static int ina219_pm_ctrl(const struct device *dev,
+	uint32_t ctrl_command,
+	enum pm_device_state *state,
+	pm_device_cb cb,
+	void *arg)
+{
+	int rc = 0;
+
+	if (ctrl_command == PM_DEVICE_STATE_SET) {
+		rc = ina219_set_power_state(dev, *state);
+	} else if (ctrl_command == PM_DEVICE_STATE_GET) {
+		rc = ina219_get_power_state(dev, state);
+	}
+
+	if (cb) {
+		cb(dev, rc, state, arg);
+	}
```

Comment:
callback has been removed, see https://github.com/zephyrproject-rtos/zephyr/pull/36757

Reply:
i like it.
I have removed the `pm_device_cb` argument as well as the if block